tests/test-recipes/metadata/_c_vendoring_detection/conda_build_config.yaml

```diff
@@ -33,8 +33,6 @@ rust_compiler:
   - rust
 rust_compiler_version:
   - 1.29.0
-CONDA_BUILD_SYSROOT:
-  - /opt/MacOSX10.10.sdk        # [osx]
```

Comment:
`CONDA_BUILD_SYSROOT` as an environment variable is easier to use as we can extract the value programmatically (see "Setup variables" in the macos job of `tests.yml`)

Reply:
How do we feel about dynamically generating `cbc.yaml`?

Reply:
Yes, that'd be exactly what I had in mind :).

Reply:
Since all of the affected recipes are built via `conda_build.api.build` I opted to create a pytest fixture that returned a `variants` dictionary. Is this sufficiently dynamic or do we want an actual `cbc.yaml` file generated? If this is sufficient we should be able to also remove the `CONDA_BUILD_SYSROOT` and `MACOSX_DEPLOYMENT_TARGET` from the macOS setup in `tests.yaml`.

Reply:
I think it should be fine to use `variants` instead of the YAML file (close enough ;).)
I'd go for setting things via env vars and then override the deployment target as variant, see https://github.com/conda/conda-build/pull/4648/files#r1038188689 .